core/src/test/java/io/questdb/cutlass/line/tcp/AlterTableLineDropActivePartitionLineTest.java

```diff
@@ -0,0 +1,189 @@
+                            "CREATE TABLE " + tableName + "( " +
+                                    "favourite_colour symbol index capacity 16, " +
+                                    "country symbol index capacity 16, " +
+                                    "orderId long, " +
+                                    "quantity int, " +
+                                    "ppu double, " +
+                                    "addressId string, " +
+                                    "timestamp timestamp" +
+                                    ") timestamp(timestamp) partition by day " +
+                                    "with maxUncommittedRows=20, commitLag=200000us" // 200 millis
+                    )
+            ) {
+                stmt.execute();
+            }
+
+            // setup a thread that will send ILP/TCP for today
+            final SOCountDownLatch ilpAgentHalted = new SOCountDownLatch(1);
+            final AtomicBoolean keepSending = new AtomicBoolean(true);
+            final AtomicLong orderId = new AtomicLong(0L);
+
+            final Thread ilpAgent = new Thread(() -> {
+                final Rnd rnd = new Rnd();
+                try (LineTcpSender sender = LineTcpSender.newSender(Net.parseIPv4("127.0.0.1"), ILP_PORT, 4 * 1024)) {
+                    while (keepSending.get()) {
+                        sender.metric(tableName)
+                                .tag("favourite_colour", rndOf(rnd, colour))
+                                .tag("country", rndOf(rnd, country))
+                                .field("orderId", orderId.getAndIncrement())
+                                .field("quantity", rnd.nextPositiveInt())
+                                .field("ppu", rnd.nextFloat())
+                                .field("addressId", rnd.nextString(50))
+                                .$();
+                        sender.flush();
+                    }
+                } finally {
+                    ilpAgentHalted.countDown();
+                }
+            });
+            ilpAgent.start();
+
+            // give the agent some time
+            while (orderId.get() < 1_000_000) {
+                Os.pause();
+            }
+
+            long beforeDropSize;
+            try (TableReader reader = serverMain.getCairoEngine().getReader(AllowAllCairoSecurityContext.INSTANCE, tableName)) {
+                beforeDropSize = reader.size();
+                Assert.assertTrue(beforeDropSize > 0L);
+            }
+
+            // drop active partition
+            try (
+                    Connection connection = DriverManager.getConnection(PG_CONNECTION_URI, PG_CONNECTION_PROPERTIES);
+                    PreparedStatement stmt = connection.prepareStatement("ALTER TABLE " + tableName + " DROP PARTITION WHERE timestamp > 0")
+            ) {
+                stmt.execute();
+            }
+
+            Os.sleep(200L); // allow a few millis more to the agent

```

Comment:
sleep based tests are non-determenistic and will flap

Lets become deteministic here please